### PR TITLE
ibc: add grpc queries for connections, channels, and clients 

### PR DESCRIPTION
**Pull request by avahowell**

### CODE REVIEW

**Comment by avahowell** [comment]
All of the previously implemented ibc queries are implemented now, this just needs to be hooked up to the grpc api

**Comment by avahowell** [comment]
Hooked up the grpc api

**Comment by hdevalence** [comment]
Looks like this is just blocked on fixing build flags for wasm?

**Comment by conorsch** [comment]
Discussed at sprint planning today: we need to land these necessary-but-not-sufficient queries to unblock a Hermes relaying spike. However, further queries are required. For Testnet 60 https://github.com/penumbra-zone/penumbra/issues/2984, we won't have full support merged, but we'll press on with support.

File: crates/core/component/ibc/src/component/grpc_query.rs
Comment (hdevalence) [inline crates/core/component/ibc/src/component/grpc_query.rs:150]: This will make the node panic if we get one of these requests, can we search/replace these `todo!()` impls with a snippet that returns an `Ok` with [`Status::unimplemented`](https://docs.rs/tonic/latest/tonic/struct.Status.html#method.unimplemented) ?

**Review by hdevalence — CHANGES_REQUESTED** [review]
Let's avoid the `todo!()`s but otherwise this LGTM!

File: crates/bin/pd/src/main.rs
Comment (hdevalence) [inline crates/bin/pd/src/main.rs:351]: Non-blocking: going forward, we want to split up the RPCs and move them into the component crates. When we do that, we'll have LOTS of different services to stitch together, and we should figure out how to cleanly add them all. Maybe `tower::Layer`s would be helpful there.

**Comment by conorsch** [comment]
We want this in for 60 (#2984). @avahowell is aiming for merge by EOD.

**Pull request merged**
